v1/protocols/ipfs.ts

```diff
@@ -122,25 +117,40 @@ export class IPFSProtocol implements Protocol<Static<typeof IPFSProtocolFields>>
   async sync (id: string, folderPath: string, options?: SyncOptions, ctx?: Ctx): Promise<Static<typeof IPFSProtocolFields>> {
     ctx?.logger.info('[ipfs] Sync Start')
     const mfsLocation = path.posix.join(this.mfsRoot, id)
-    const mfsSyncOptions = {
-      noDelete: options?.ignoreDeletes ?? false
-    }
-    if (this.mfsSync === null || this.ipfs === null) {
-      throw new Error('MFS must be initialized using load() before calling sync()')
+    if (this.ipfs === null) {
+      throw new Error('IPFS must be initialized using load() before calling sync()')
     }
 
-    // Make a folder in MFS
-    await this.ipfs.files.mkdir(mfsLocation, {
-      parents: true,
-      flush: true,
+    let lastEntry = null
+
+    const glob = IPFS.globSource(folderPath, '**/*')
+    const files = this.ipfs.addAll(glob, {
+      pin: false,
+      wrapWithDirectory: true,
       cidVersion: 1
     })
 
-    // Sync local disk files to MFS
-    for await (const change of this.mfsSync.fromFSToMFS(folderPath, mfsLocation, mfsSyncOptions)) {
-      ctx?.logger.debug(change)
+    for await (const file of files) {
+      ctx?.logger.debug(`[ipfs] added ${file.path}`)
+      lastEntry = file
+    }
+
+    if (lastEntry == null) {
+      throw new Error('No files found')
```

Comment:
Why is this an error? Wouldn't our 'delete' be the equivalent of syncing an empty set of files?

Reply:
Honestly, I'm not certain this condition could even be reached. But that's a good point, we should publish an empty dir if there was no files added.

Reply:
I'll make an issue to track this unless you think it should be a blocker for the PR

Reply:
Cool, now it publishes an empty CID if there's no files added.